Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -18,38 +35,457 @@
+    AZStd::vector<AzToolsFramework::ActionOverride> TransformMode::PopulateActions(
+        [[maybe_unused]] const AZ::EntityComponentIdPair& entityComponentIdPair)
     {
         return {};
     }
 
+    void TransformMode::DrawFace(
+        AzFramework::DebugDisplayRequests& debugDisplay, WhiteBoxMesh* mesh, const Api::PolygonHandle& polygon, const AZ::Color& color)
+    {
+        AZStd::vector<AZ::Vector3> triangles = Api::PolygonFacesPositions(*mesh, polygon);
+        const AZ::Vector3 normal = Api::PolygonNormal(*mesh, polygon);
+
+        if (!triangles.empty())
+        {
+            // draw selected polygon
+            debugDisplay.PushMatrix(AZ::Transform::CreateTranslation(normal * ed_whiteBoxPolygonViewOverlapOffset));
+
+            debugDisplay.SetColor(color);
+            debugDisplay.DrawTriangles(triangles, color);
+
+            debugDisplay.PopMatrix();
+        }
+    }
+
+    void TransformMode::DrawOutline(
+        AzFramework::DebugDisplayRequests& debugDisplay, WhiteBoxMesh* mesh, const Api::PolygonHandle& polygon, const AZ::Color& color)
+    {
+        Api::VertexPositionsCollection outlines = Api::PolygonBorderVertexPositions(*mesh, polygon);
+        if (!outlines.empty())
+        {
+            // draw outline
+            debugDisplay.DepthTestOn();
+            debugDisplay.SetColor(color);
+            debugDisplay.SetLineWidth(cl_whiteBoxEdgeVisualWidth);
+            for (const auto& outline : outlines)
+            {
+                debugDisplay.DrawPolyLine(outline.data(), aznumeric_caster(outline.size()));
+            }
+            debugDisplay.DepthTestOff();
+        }
+    }
+
     void TransformMode::Display(
         [[maybe_unused]] const AZ::EntityComponentIdPair& entityComponentIdPair,
-        [[maybe_unused]] const AZ::Transform& worldFromLocal,
-        [[maybe_unused]] const IntersectionAndRenderData& renderData,
-        [[maybe_unused]] const AzFramework::ViewportInfo& viewportInfo,
-        [[maybe_unused]] AzFramework::DebugDisplayRequests& debugDisplay)
+        const AZ::Transform& worldFromLocal,
+        const IntersectionAndRenderData& renderData,
+        const AzFramework::ViewportInfo& viewportInfo,
+        AzFramework::DebugDisplayRequests& debugDisplay)
     {
+        WhiteBoxMesh* whiteBox = nullptr;
+        EditorWhiteBoxComponentRequestBus::EventResult(
+            whiteBox, m_entityComponentIdPair, &EditorWhiteBoxComponentRequests::GetWhiteBoxMesh);
+
+        debugDisplay.PushMatrix(worldFromLocal);
+
+        if(m_polygonIntersection.has_value()) {
+            auto& polygonIntersection = m_polygonIntersection.value();
+            DrawFace(debugDisplay, whiteBox, polygonIntersection.GetHandle(), cl_whiteBoxPolygonHoveredColor);
+            DrawOutline(debugDisplay, whiteBox, polygonIntersection.GetHandle(), cl_whiteBoxPolygonHoveredOutlineColor);
```

Comment:
I don't really like how these properties are by manipulator action. could we define a set of fixed colors for selection/selected for the mesh outline, vertices and edges. I don't imagine the colors should vary if much at all between all these modes.